tests/rbd/test_rbd_persistent_writeback_cache_invalidate.py

```diff
@@ -0,0 +1,159 @@
+"""RBD Persistent write back cache, Test cache invalidate at
+client, pool and image level
+
+Test Case Covered:
+CEPH-83574709 - Verify persistent write back cache invalidate at
+image/pool/client(node) level in SSD mode
+
+Steps :
+1) Setup persistent write cache for an image at client, pool and image level
+2) Write data to the image using fio jobs
+3) While IOs are running, issue invalidate cache and check cache status when
+Ios started and after issuing command check PWL cache file is deleted in the
+mounted path , cache feature is disabled in cache status
+
+Environment and limitations:
+ - The cluster should have 5 nodes + 1 SSD cache node
+ - cluster/global-config-file: config/quincy/upi/octo-5-node-env.yaml
+ - Should be Bare-metal.
+
+Support
+- Configure cluster with PWL Cache.
+- Only replicated pool supported, No EC pools.
+"""
+from time import sleep
+
+from ceph.parallel import parallel
+from ceph.utils import get_node_by_id
+from tests.rbd.rbd_peristent_writeback_cache import (
+    PersistentWriteAheadLog,
+    fio_ready,
+    get_entity_level,
+)
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def validate_cache_invalidate(cache, cfg, client):
+    """
+    Perform cache invalidate and verify the results
+
+    Args:
+        cache: PersistentWriteAheadLog object
+        cfg: test config
+        client: cache client node
+    """
+    out = cache.get_image_cache_status(cfg["image_spec"])
+    log.debug(f"RBD status of image: {out}")
+
+    cache_file = out.get("persistent_cache", {}).get("path")
+    cache_file = cache_file.replace(f"{cache.pwl_path}/", "")
+    sleep(60)
+    out, err = cache.invalidate(cfg["image_spec"])
+    if not out:
+        log.info(f"Cache invalidate for image {cfg['image_spec']} is success")
+    else:
+        log.error(
+            f"Cache invalidate failed for image {cfg['image_spec']} with error {err}"
+        )
+        raise Exception(
+            f"Cache flush invalidate for image {cfg['image_spec']} with error {err}"
+        )
+
+    # Verify that the cache file is removed, cache feature is disabled in cache status
+    cache_present = False
+    try:
+        cache.check_cache_file_exists(
+            cfg["image_spec"],
+            20,
+            **cfg,
+        )
+        log.error("Cache file is present even after cache invalidate")
+        cache_present = True
+    except Exception:
+        log.info("Cache file is removed after cache invalidate")
+
+    if cache_present:
+        raise Exception(
```

Comment:
this exception `try` block is missing.

Reply:
I didn't understand your comment. This exception is not part of try block. It is supposed to be outside try block and I am raising the exception.